simpleviewinc/keg-herkin
# Part 7: Test Results - Cmd and  Report display

## Context

* The Test Results tab needs some cleanup

## Goal

* Clean up the Test Results tab

## Updates

* Fixed test reports loading
* Fixed issue with backend feature parsing
* Added action to load a test file, based on clicked on test file
  * This then allows loading the latest test results for a test file from the side menu
* Minor cleanup to the cmd output
* Added toggle loading the test-results iframe based on if tests are running
  * Once they are done running, the test-results iframe loads the latest test-results file
* Fixed issue with reports paths
  * This is an odd issue, in the past I needed quotes around the string
    * That does not seem to be the case any longer
    * **This may need to be investigated more**

## Testing
**Setup**
* Move to the keg-herkin local repo and pull down these changes with `keg pr <>`
* Switch to tap-events-force, and pull down this branch => `test-feature-files`
  * Cmd => `keg evf && keg fetch && keg br test-feature-files` 
* While cwd is the tap-events-force repo, start keg-herkin with the following cmd 
  * `keg evf && keg herkin start --from keg-herkin:cmd-report-loading`
* Once it starts navigate to http://herkin-cmd-report-loading.local.kegdev.xyz/ in your browser
* The test settings modal should display
  * Select the `dayNavigation.feature` from the drop down
  * Then click `Do It`
* The feature should load on the page along with a list of all the Definitions

**Test 1**
* Click the Run Tests Button
* This will run the test from the feature
* It should automatically switch to the `Test Results` tab
  * The output from the running test command should display in top `Test Output` section
    * Looks almost like a terminal
  * The `Test Results` Section below it should show a message; `Test Running` with a spinner
  * Once the test finish, the `Test Results Report` should display in the `Test Results` section

> * **IMPORTANT** - At this point it does not matter if the tests fail or not.
> * So don't worry about it when the feature fails
> * If you really care about seeing all green, select the other feature file `Scrollserssions.feature`
>   * Then run the tests on it, that one should pass.
>   * And NO, I don't care that it's spelled `serssions`, that is not the point
  
**Test 2**
* At the bottom of the `Test Results` tab, there should be a `Re-Run Tests` button.
* After running the tests once for a file, click that button, and it should re-run the tests again

**Test 3**
* Doing **Test 1/2** are required before doing this one
* Open the left-hand side menu
* You should see a Reports, section, click on it, then click on the feature sub-section
* Inside the feature sub-section, you'll see `test results` files
* Click on one, and ensure it opens up the test-results section with that feature file

> **NOTES** - It's important to note, that it does not actually open the `results file` you clicked on
> It will open the **most** recent `test results` for the `test file` **related** to the `results file` you clicked on
> But because at this time `feature files` can have **one** test-results file, it ends up being the one you clicked on
> This will change in the future where it will load the actual file you click, but for now, the end result is the same

**Test 4**
* Open the Events-Force Tap in your editor
* Navigate to tests/bdd/steps, and add a new step definition
  * The type and what it does, doesn't matter, it just needs to be valid
  * Something like this is fine => 
  ```js
    const { When } = require('HerkinParkin')
    When('I test a thing', () => {})
  ```
  * Just make sure the matcher string is different then any other definitions
  * Create a new file for it, or put it in an existing definition file, it should not matter
* Once the step definition is created, go back to keg-herkin in the browser
  * In the right side of the `Editor Tab`, you should see the Definitions list
    * On the top right, you will see a new `sync` text and icon
    * Click the sync button, which should cause the definition to reload
    * Ensure the new definition you created, shows up in the list
